packages/datadog-instrumentations/src/mocha.js

```diff
@@ -172,8 +175,10 @@ function mochaHook (Runner) {
       if (!asyncResource) {
         asyncResource = new AsyncResource('bound-anonymous-fn')
         testFileToSuiteAr.set(suite.file, asyncResource)
+        const isUnskippable = unskippableSuites.includes(suite.file)
+        const isForcedToRun = isUnskippable && suitesToSkip.includes(getTestSuitePath(suite.file, process.cwd()))
         asyncResource.runInAsyncScope(() => {
-          testSuiteStartCh.publish(suite)
+          testSuiteStartCh.publish({ testSuite: suite.file, isUnskippable, isForcedToRun })
         })
```

Comment:
Noob question, is this called multiple times? and that's the reason to do this just to create everything on the first call?

Reply:
yeah, there's no concept of "test file" (what we call suite) in mocha, so we have to detect when we are in the first mocha's `suite` to start our own suite. It's a bit hacky but it's the only way to keep consistency across frameworks